src/App.jsx

```diff
@@ -63,6 +64,7 @@ function App() {
                 ? Object.keys(data).map((key) => (
                   <li onClick={() => {
                     setActiveFileId(key)
+                    setTabName(data[key].name)
```

Comment:
I don't like this, because it duplicates state.

Think about it like this: `activeFileId` and `tabName` _always_ change together.

I would suggest to just use `activeFileId` to inform `tabName` by computing a derived value rather than introducing a new piece of state.
```suggestion
```